rsconnect/main.py

```diff
@@ -101,7 +106,7 @@
 def cli_exception_handler(func: Callable[P, T]) -> Callable[P, T]:
     @wraps(func)
     def wrapper(*args: P.args, **kwargs: P.kwargs):
-        def failed(err: str):
+        def failed(err: str) -> Never:
```

Comment:
The explicit `Never` return helps pyright recognize that the calls to `failed()` will never return.